src/js/components/Button/StyledButtonKind.js

```diff
@@ -80,13 +80,16 @@ const basicStyle = (props) => css`
 
   // when button has badge, the SVG won't necessarily
   // be the direct descendant
-  ${props.badge
-    ? `
-  svg {
+  ${(props.badge &&
+    `svg {
     vertical-align: bottom;
-  }`
-    : `> svg {
+  }`) ||
+  (props.icon?.props?.size !== 'small' &&
```

Comment:
This feels a bit fragile. The icon could have `size="13px"`.

One idea: `useLayoutEffect` to measure the actual height of the icon after rendering and adjust as needed.

Another idea: change the `<button />` to use `display: flex` and align items to the center.

I'm trying to think if there's a way to manage the layout without `vertical-align` and needing to have CSS selectors  for child elements.

Reply:
Sounds good, looking into it...